Dungeons/Dungeons/CorridorNodeLayouter.cs

```diff
@@ -95,136 +93,42 @@ public DungeonLevel DoLayout(List<DungeonNode> nodes, LayouterOptions opt = null
     protected virtual void LayoutNodes(DungeonNode level, List<DungeonNode> mazeNodes)
     {
       this.mazeNodes = mazeNodes;
-      var random = RandHelper.Random;
-      const int centralRoomPosition = 12;
-      int[] wallLenght = { 0, 1, 2, 3 };
-      List<int[]> positionsOfNodes = new List<int[]>();
-
-      var currentWidth = 0;
-      var currentHeight = 0;
-      List<int> distances = new List<int>();
+      var d = new DungeonGenerator(container);
+      var infoC = new GenerationInfo();
+      infoC.NumberOfRooms = 1;
+
+      mazeNodes[0].Placement = RoomPlacement.LeftUpper;
+      mazeNodes[1].Placement = RoomPlacement.RightUpper;
+      mazeNodes[2].Placement = RoomPlacement.RightLower;
+      mazeNodes[3].Placement = RoomPlacement.LeftLower;
+      mazeNodes[4].Placement = RoomPlacement.Center;
+
       mazeNodes[0].GenerateDoors(RoomPlacement.LeftUpper);
       mazeNodes[1].GenerateDoors(RoomPlacement.LeftLower);
       mazeNodes[2].GenerateDoors(RoomPlacement.RightUpper);
       mazeNodes[3].GenerateDoors(RoomPlacement.RightLower);
       mazeNodes[4].GenerateDoors(RoomPlacement.Center);
-      for (int i = 0; i < mazeNodes.Count; i++) //creating rooms
-      {
-        if (i == (int)RoomPlacement.LeftUpper || i == (int)RoomPlacement.RightUpper)
-        {
-          positionsOfNodes.Add(new int[]{ currentWidth,currentHeight});
-          currentHeight += random.Next(24, 28);
-          distances.Add(currentHeight);
-        }
-        if (i == (int)RoomPlacement.LeftLower || i == (int)RoomPlacement.RightLower)
-        {
-          if (i == (int)RoomPlacement.RightLower)
-            currentWidth = random.Next(24, 28);
-          positionsOfNodes.Add(new int[] { currentWidth, currentHeight });
-          currentHeight = 0;
-          if (currentWidth == 0)
-            currentWidth += random.Next(24, 28);
-          distances.Add(currentWidth);
-        }
-      }
-      positionsOfNodes.Add(new int[] { centralRoomPosition, centralRoomPosition });
 
-
-      var d = new DungeonGenerator(container);
-      var infoC = new GenerationInfo();
-      infoC.NumberOfRooms = 1;
-      currentWidth = 0;
-      currentHeight = 0;
-      var h = 0;
-      var h1 = 0;
-      for (int i = 0; i < 4; i++) //creating corrindors
+      var conn = new DungeonNodeConnector(); //generating normal rooms
+      for (int i = 0; i < 4; i++)
```

Comment:
Jesli mozna prosci to fajnie by bylo przerobic ten fragment tak zeby byla mozliwosc wynegerowania prostszego dungeonu. Np tylko dwa pokoje (bez centralnego).